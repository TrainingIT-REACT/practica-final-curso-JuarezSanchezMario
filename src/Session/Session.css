.Session {
  padding:5%;
}

/*.App h1 {
  text-align: center;
}*/
