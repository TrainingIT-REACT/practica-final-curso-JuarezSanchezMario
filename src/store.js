import { createStore, applyMiddleware, combineReducers } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';


// Reducers
import albums from "./reducers/albums";
import historico from "./reducers/historico";
import songs from "./reducers/songs";
import user from "./reducers/user";

export const initialState = {
  albums: {
    albums: [
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 2,
        name: 'Battle grounds',
        artist: 'Florian',
        cover: '/images/cover.jpg'
      },
      {
        id: 3,
        name: 'Decisions decisions',
        artist: 'Skylar',
        cover: '/images/cover.jpg'
      },
      {
        id: 4,
        name: 'Blank canvas',
        artist: 'Skylar',
        cover: '/images/cover.jpg'
      },
      {
        id: 5,
        name: 'Ice cold',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 6,
        name: 'Honesty',
        artist: 'Ted Garrett',
        cover: '/images/cover.jpg'
      },
      {
        id: 7,
        name: 'Creative director',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 8,
        name: 'No guarantees',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 9,
        name: 'Don\'t push this button',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 10,
        name: 'Cherry',
        artist: 'Stacy',
        cover: '/images/cover.jpg'
      }
    ],
    loading: false,
    recomendados: [
      {
        id: 8,
        name: 'No guarantees',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 9,
        name: 'Don\'t push this button',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 7,
        name: 'Creative director',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 2,
        name: 'Battle grounds',
        artist: 'Florian',
        cover: '/images/cover.jpg'
      }
    ],
    selectedAlbum: {
      id: 5,
      name: 'Ice cold',
      artist: 'Kimberley',
      cover: '/images/cover.jpg'
    }
  },
  historico: {
    albums: [
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 5,
        name: 'Ice cold',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      }
    ],
    songs: [
      {
        id: 1,
        name: 'Take On Me',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 1
      },
      {
        id: 2,
        name: 'S.O.S.',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 1
      }
    ]
  },
  songs: {
    loading: false,
    songs: [
      {
        id: 21,
        name: 'Back In Black',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 22,
        name: 'Rock and Roll Ain\'t Noise Pollution',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 23,
        name: 'Dirty Deeds Done Dirt Cheap',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 24,
        name: 'Whole Lotta Rosie',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 25,
        name: 'Let There Be Rock',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      }
    ]
  },
  user: {
    name: 'Mario',
    email: 'mjs.mariojuarezsanchez@gmail.com',
    logged: true
  }
}

export default createStore(
  combineReducers({albums, historico, songs, user}),
  composeWithDevTools(applyMiddleware(thunk))
);
