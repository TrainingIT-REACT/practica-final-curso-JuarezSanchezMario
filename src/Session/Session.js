import React, { Component } from 'react';
// Css
import './Session.css';

class Session extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: true,
      albums: []
    }
  }

  componentDidMount() {

  }

  render() {
    return (
      <div className="Perfil">
        
      </div>
    );
  }
}

export default Session;
