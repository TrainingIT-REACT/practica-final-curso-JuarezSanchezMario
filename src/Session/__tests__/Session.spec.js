import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { initialState } from "../../store";
import { Router } from "react-router-dom";
import thunk from "redux-thunk";
import history from "../../App/history";

// Componente
import Session from "../Session";

describe("Session", () => {

  let store, wrapper;
const mockStore = configureStore([thunk]);

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <Session />
        </Router>
      </Provider>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
