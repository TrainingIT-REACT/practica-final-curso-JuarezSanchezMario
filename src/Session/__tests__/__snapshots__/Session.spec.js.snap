// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "vakbp5",
          "pathname": "/perfil",
          "search": "",
          "state": Object {
            "from": "inicio",
          },
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(Session)>
      <Session
        dispatchUser={[Function]}
        user={
          Object {
            "email": "mjs.mariojuarezsanchez@gmail.com",
            "logged": true,
            "name": "Mario",
          }
        }
      >
        <div
          className="Session"
        >
          <Badge
            pill={false}
            variant="secondary"
          >
            <span
              className="badge badge-secondary"
            >
              Inicio sesión
            </span>
          </Badge>
          <Form
            inline={false}
            onSubmit={[Function]}
          >
            <form
              className=""
              onSubmit={[Function]}
            >
              <FormGroup
                controlId="name"
              >
                <div
                  className="form-group"
                >
                  <FormControl
                    placeholder="Name"
                    required={true}
                    type="text"
                  >
                    <input
                      className="form-control"
                      id="name"
                      placeholder="Name"
                      required={true}
                      type="text"
                    />
                  </FormControl>
                </div>
              </FormGroup>
              <FormGroup
                controlId="email"
              >
                <div
                  className="form-group"
                >
                  <FormControl
                    placeholder="Email"
                    required={true}
                    type="email"
                  >
                    <input
                      className="form-control"
                      id="email"
                      placeholder="Email"
                      required={true}
                      type="email"
                    />
                  </FormControl>
                </div>
              </FormGroup>
              <FormGroup
                controlId="pass"
              >
                <div
                  className="form-group"
                >
                  <FormControl
                    placeholder="Contraseña"
                    required={true}
                    type="password"
                  >
                    <input
                      className="form-control"
                      id="pass"
                      placeholder="Contraseña"
                      required={true}
                      type="password"
                    />
                  </FormControl>
                </div>
              </FormGroup>
              <Button
                active={false}
                disabled={false}
                type="submit"
                variant="primary"
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="submit"
                >
                  LOG IN
                </button>
              </Button>
            </form>
          </Form>
          <Redirect
            push={false}
            to={
              Object {
                "pathname": "/perfil",
                "state": Object {
                  "from": "inicio",
                },
              }
            }
          />
        </div>
      </Session>
    </Connect(Session)>
  </Router>
</Provider>
`;
